### DISCUSSION TITLE
More comprehensive guide for setting up Immich?

### DISCUSSION THREAD

**Discussion by Cr0vis** [immich-app/immich#42] [General]
Any chance you could link to a more step by step, easier to understand guide for setting up the Immich server code stuff. I myself don’t have any knowledge on coding and docker stuff but want to test this out. 

Or would you rather me wait until a stable version Is out that’s more easier to use. If so do you have a rough eta on when one would or could come out?

Thank ya in advance

**Comment by alextran1502** [comment]
Hey, I was thinking of making more detail instruction in the Wiki as well. It will be the same method regardless of the state of the application. I can help you get started if you would like!

Can you tell me which environment are you planning to run on?

**Reply by Cr0vis** [reply]
Yea, if you can help me get it started up I would appreciate it

I’m using vs code with windows 11

**Reply by alextran1502** [reply]
Can you follow this guide and see if you can install Docker on Windows https://docs.docker.com/desktop/windows/install/

**Reply by Cr0vis** [reply]
I’ve already downloaded it, correctly I hope

**Reply by alextran1502** [reply]
From the command line, can you run

docker run hello-world

And verify that it works

**Reply by Cr0vis** [reply]
Seems like it works, I got this

![image](https://user-images.githubusercontent.com/99766020/158022783-160ffff1-cf1f-4958-9d5f-fa75fb979d50.png)

**Reply by alextran1502** [reply]
Perfect!

Now can you clone/download the repository? Then, follow the steps in the readme file and let me know if you run into any problems along the way

**Reply by Cr0vis** [reply]
okay, if I run into any problems I'll report back here

**Reply by Cr0vis** [reply]
I got this, I have no clue what it means tho; something went wrong
![Screenshot 2022-03-12 104535](https://user-images.githubusercontent.com/99766020/158024890-1171654f-cc56-47f3-93ee-a0084611b3a2.png)
![Screenshot 2022-03-12 104615](https://user-images.githubusercontent.com/99766020/158024892-e9636cc2-e4eb-4482-94be-88697f8fb677.png)

**Reply by alextran1502** [reply]
You will need to supply the value for most of the key in `.env` file. Here is a working example you can try

```bash
# STAGE
NODE_ENV=development

# Database
DB_USERNAME=postgres
DB_PASSWORD=postgres
DB_DATABASE_NAME=immich

# Upload File Config
UPLOAD_LOCATION=C:\Users\Aueso\Desktop\Photo Backup

# JWT SECRET
JWT_SECRET=secret

# MAPBOX
## ENABLE_MAPBOX is either true of false -> if true, you have to provide MAPBOX_KEY
ENABLE_MAPBOX=false
MAPBOX_KEY=
```

I suggest for `UPLOAD_LOCATION` you create a folder without white space, something like PhotoBackup. 

Is there any more information below the error message you sent earlier?

**Reply by Cr0vis** [reply]
Not really, here's the full error message. I'll come back after trying what you have told me to do
![2](https://user-images.githubusercontent.com/99766020/158031126-8fbd95d2-32f5-47bf-81f8-802b12ad5a80.jpg)

**Reply by Cr0vis** [reply]
Now I'm getting this after following what you said, and running VS Code as administrator
![image](https://user-images.githubusercontent.com/99766020/158032393-95567138-3142-4092-b8e1-d95e39461044.png)

**Reply by alextran1502** [reply]
Hmm this is interesting, the error showed that there was a network connection issue.  Can you try again if this is still there?

**Reply by Cr0vis** [reply]
Okay, now i'm getting this
![image](https://user-images.githubusercontent.com/99766020/158040223-da102d30-3d33-46e7-a82a-c7e4092513a8.png)

**Reply by alextran1502** [reply]
This error seems to be specifically for Windows. Can you try to enable WSL on Windows? There is a guide [here](https://pureinfotech.com/install-wsl-windows-11/). Docker seems to be problematic on Windows, but it works very well on Linux. WSL will help you to run a Linux kernel inside Windows

**Reply by Cr0vis** [reply]
I have wsl 2 installed and working with docker; I kind of gave up on it at this point, but thanks for your time and help. 

Just doesn’t seem to work with on windows.

**Reply by Cr0vis** [reply]
Do I need to install a different Linux distro or is the default one that comes with docker good?

**Reply by alextran1502** [reply]
So you installed WSL2 and Docker on WSL2 and the application works, correct?

For your second question, the Linux distro comes with WSL is ubuntu I believe and it should work with no problem

**Comment by Mazpaz** [comment]
Hi I don't have much knowledge of coding, I have downloaded the app on my iPhone but I don't know how to sign up?. if you could please help me with that thanks
